cmake/pika_check_cxx_standard.cmake

```diff
@@ -31,11 +31,5 @@ endif()
 set(CMAKE_CXX_STANDARD ${PIKA_WITH_CXX_STANDARD})
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
-# We explicitly set the default to 98 to force CMake to emit a -std=c++XX flag.
-# Some compilers (clang) have a different default standard for cpp and cu files,
-# but CMake does not know about this difference. If the standard is set to the
-# .cpp default in CMake, CMake will omit the flag, resulting in the wrong
-# standard for .cu files.
-set(CMAKE_CXX_STANDARD_DEFAULT 98)
```

Comment:
I'd like to keep these around. Even though we explicitly now use clang as  `CUDA` compiler, NVHPC is still a `CXX`-only compiler but it still compiles `.cu` files.

Ignoring any issues with CMake doing the wrong thing, I still like to have the flag explicitly emitted. Explicit is usually better than implicit.